zcash_proofs/src/sapling/prover.rs

```diff
@@ -56,27 +55,22 @@ impl SaplingProvingContext {
         merkle_path: MerklePath<Node>,
         proving_key: &Parameters<Bls12>,
         verifying_key: &PreparedVerifyingKey<Bls12>,
-    ) -> Result<(Proof<Bls12>, jubjub::ExtendedPoint, PublicKey), ()> {
+    ) -> Result<(Proof<Bls12>, ValueCommitment, PublicKey), ()> {
         // Initialize secure RNG
         let mut rng = OsRng;
 
         // We create the randomness of the value commitment
-        let rcv = jubjub::Fr::random(&mut rng);
+        let rcv = ValueCommitTrapdoor::random(&mut rng);
 
         // Accumulate the value commitment randomness in the context
-        {
-            let mut tmp = rcv;
-            tmp.add_assign(&self.bsk);
-
-            // Update the context
-            self.bsk = tmp;
-        }
+        self.bsk += &rcv;
 
         // Construct the value commitment
-        let value_commitment = ValueCommitment {
+        let value_commitment_inputs = crate::circuit::sapling::ValueCommitmentPreimage {
```

Comment:
Consider renaming to `value_commitment_opening`.

Reply:
Also I suggest importing `ValueCommitmentPreimage` (or `ValueCommitmentOpening` if it's renamed). It is used more tab once in the file.

Reply:
The reason for the long path import is that this was originally named `ValueCommitment` and there were two of them. We can fix that now.